pypeit/images/combineimage.py

```diff
@@ -133,112 +127,77 @@ def run(self, bias=None, scattlight=None, flatimages=None, ignore_saturation=Fal
                 img_stack = np.zeros(shape, dtype=float)
                 scl_stack = np.ones(shape, dtype=float)
                 rn2img_stack = np.zeros(shape, dtype=float)
                 basev_stack = np.zeros(shape, dtype=float)
                 gpm_stack = np.zeros(shape, dtype=bool)
-                lampstat = [None]*self.nfiles
-                exptime = np.zeros(self.nfiles, dtype=float)
+                lampstat = [None]*self.nimgs
+                exptime = np.zeros(self.nimgs, dtype=float)
 
             # Save the lamp status
             # TODO: As far as I can tell, this is the *only* place rawheadlist
             # is used.  Is there a way we can get this from fitstbl instead?
-            lampstat[kk] = self.spectrograph.get_lamps_status(pypeitImage.rawheadlist)
+            lampstat[kk] = self.spectrograph.get_lamps_status(rawImage.rawheadlist)
             # Save the exposure time to check if it's consistent for all images.
-            exptime[kk] = pypeitImage.exptime
+            exptime[kk] = rawImage.exptime
             # Processed image
-            img_stack[kk] = pypeitImage.image
+            img_stack[kk] = rawImage.image
             # Get the count scaling
-            if pypeitImage.img_scale is not None:
-                scl_stack[kk] = pypeitImage.img_scale
+            if rawImage.img_scale is not None:
+                scl_stack[kk] = rawImage.img_scale
             # Read noise squared image
-            if pypeitImage.rn2img is not None:
-                rn2img_stack[kk] = pypeitImage.rn2img * scl_stack[kk]**2
+            if rawImage.rn2img is not None:
+                rn2img_stack[kk] = rawImage.rn2img * scl_stack[kk]**2
             # Processing variance image
-            if pypeitImage.base_var is not None:
-                basev_stack[kk] = pypeitImage.base_var * scl_stack[kk]**2
+            if rawImage.base_var is not None:
+                basev_stack[kk] = rawImage.base_var * scl_stack[kk]**2
             # Final mask for this image
             # TODO: This seems kludgy to me. Why not just pass ignore_saturation
             # to process_one and ignore the saturation when the mask is actually
             # built, rather than untoggling the bit here?
             if ignore_saturation:  # Important for calibrations as we don't want replacement by 0
-                pypeitImage.update_mask('SATURATION', action='turn_off')
+                rawImage.update_mask('SATURATION', action='turn_off')
             # Get a simple boolean good-pixel mask for all the unmasked pixels
-            gpm_stack[kk] = pypeitImage.select_flag(invert=True)
+            gpm_stack[kk] = rawImage.select_flag(invert=True)
+            file_list.append(rawImage.filename)
 
+        # TODO JFH: This should not be here, but rather should be done in the Calibrations class. Putting in calibration specific
+        # lamp checking in an image combining class is not ideal. 
```

Comment:
In principle, I agree.  A lamp check that is specific to arcs is out of place in a generic image-combining class.  We could push the lamp check into `Calibrations.get_arc`, but we need to be careful about the amount of code that would need to be repeated and minimize the number of times we need to open the raw files.

I would say that the exposure time check should stay here (and we should deal with combining images with different exposure times better...).

Reply:
I agree. In fact, since we are now passing in a `pypeitImage` (and not doing the reading/processing in this function as before), we should move the lamp checks to `pypeit/images/buildimage.buildimage_fromlist()`. This allows @jhennawi to remove the fudged function in `jwst_nirspec()` I think this makes the most sense.

I mostly agree with @kbwestfall that the exposure time stuff should stay here though, although I could be convinced either way, and that might be worth a separate discussion.

Reply:
@rcooke-ast this arc lamp checking stuff was not added by me in this PR, but rather I think by you in a different context. Can you please confirm?  All I did was move that bit of code here from another of the image combine routines in order to be compatible with the refactor.

Reply:
It's a simple change... I will submit a small PR into this so you can see what I mean. I just didn't want to monkey around with your PR.